### PR TITLE
[Doc] Update documentation for term: "tensor" -> "field" by @Rullec

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #1500

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
Enjoyed the new workflow :)

### CODE REVIEW

**Comment by Rullec** [comment]
Thanks for this new invention! I only have a suggestion:
Maybe I still need to merge the lastest master into `rullec-field`, and then do some replacement for some recent changes?

File: docs/compilation.rst [hunk 1]
```diff
@@ -21,17 +21,17 @@ Let's consider the following simple kernel:
...
 
 
-We allocate two 1D tensors to simplify discussion:
+We allocate two 1D fields to simplify discussion:
 
 .. code-block:: python
 
-  x = ti.var(dt=ti.f32, shape=128)
-  y = ti.var(dt=ti.f32, shape=16)
+  x = ti.field(dt=ti.f32, shape=128)
```
Comment (Rullec) [inline docs/compilation.rst:33]:
```suggestion
  x = ti.field(dtype=ti.f32, shape=128)
  y = ti.field(dtype=ti.f32, shape=16)
```

File: docs/compilation.rst [hunk 2]
```diff
@@ -21,17 +21,17 @@ Let's consider the following simple kernel:
...
+  y = ti.field(dt=ti.f32, shape=16)
```
Comment (Rullec) [inline docs/compilation.rst:34]:
```suggestion
  x = ti.field(dtype=ti.f32, shape=128)
  y = ti.field(dtype=ti.f32, shape=16)
```

File: docs/gui.rst [hunk 3]
```diff
@@ -401,7 +401,7 @@ Image I/O
 
         shape = (512, 512)
         type = ti.u8
-        pixels = ti.var(dt=type, shape=shape)
+        pixels = ti.field(dt=type, shape=shape)
```
Comment (Rullec) [inline docs/gui.rst:404]:
```suggestion
        pixels = ti.field(dtype=type, shape=shape)
```

File: docs/hello.rst [hunk 4]
```diff
@@ -17,7 +17,7 @@ Running the Taichi code below (``python3 fractal.py`` or ``ti example fractal``)
     ti.init(arch=ti.gpu)
 
     n = 320
-    pixels = ti.var(dt=ti.f32, shape=(n * 2, n))
+    pixels = ti.field(dt=ti.f32, shape=(n * 2, n))
```
Comment (Rullec) [inline docs/hello.rst:20]:
```suggestion
    pixels = ti.field(dtype=ti.f32, shape=(n * 2, n))
```

File: docs/hello.rst [hunk 5]
```diff
@@ -95,19 +95,19 @@ Taichi programs run on either CPUs or GPUs. Initialize Taichi according to your
...
+(Sparse) fields
+---------------
 
-Taichi is a data-oriented programming language where dense or spatially-sparse tensors are the first-class citizens.
-See :ref:`sparse` for more details on sparse tensors.
+Taichi is a data-oriented programming language where dense or spatially-sparse fields are the first-class citizens.
+See :ref:`sparse` for more details on sparse fields.
 
-In the code above, ``pixels = ti.var(dt=ti.f32, shape=(n * 2, n))`` allocates a 2D dense tensor named ``pixels`` of
+In the code above, ``pixels = ti.field(dt=ti.f32, shape=(n * 2, n))`` allocates a 2D dense field named ``pixels`` of
```
Comment (Rullec) [inline docs/hello.rst:110]:
```suggestion
In the code above, ``pixels = ti.field(dtype=ti.f32, shape=(n * 2, n))`` allocates a 2D dense field named ``pixels`` of
```

**Comment by archibate** [comment]
> Thanks for this new invention! I only have a suggestion:
> Maybe I still need to merge the lastest master into `rullec-field`, and then do some replacement for some recent changes?

We don't need if the changeset is small and minor.

File: docs/offset.rst [hunk 6]
```diff
@@ -3,14 +3,14 @@
 Coordinate offsets
 ==================
 
-- A Taichi tensor can be defined with **coordinate offsets**. The offsets will move tensor bounds so that tensor origins are no longer zero vectors. A typical use case is to support voxels with negative coordinates in physical simulations.
+- A Taichi field can be defined with **coordinate offsets**. The offsets will move field bounds so that field origins are no longer zero vectors. A typical use case is to support voxels with negative coordinates in physical simulations.
 - For example, a matrix of ``32x64`` elements with coordinate offset ``(-16, 8)`` can be defined as the following:
 
 .. code-block:: python
 
     a = ti.Matrix(2, 2, dt=ti.f32, shape=(32, 64), offset=(-16, 8))
```
Comment (Rullec) [inline docs/offset.rst:11]:
```
    a = ti.Matrix(2, 2, dtype=ti.f32, shape=(32, 64), offset=(-16, 8))
```

File: docs/offset.rst [hunk 7]
```diff
@@ -19,14 +19,14 @@ In this way, the tensor's indices are from ``(-16, 8)`` to ``(16, 72)`` (exclusi
...
-    c = ti.Matrix(2, 1, dt=ti.f32, shape=None, offset=(32,))             # AssertionError
-    d = ti.Matrix(3, 2, dt=ti.f32, shape=(32, 32), offset=(-16, ))       # AssertionError
-    e = ti.var(dt=ti.i32, shape=16, offset=-16)                          # Works!
-    f = ti.var(dt=ti.i32, shape=None, offset=-16)                        # AssertionError
-    g = ti.var(dt=ti.i32, shape=(16, 32), offset=-16)                    # AssertionError
+    a = ti.Matrix.field(2, 3, dtype=ti.f32, shape=(32,), offset=(-16, ))          # Works!
+    b = ti.Vector.field(3, dtype=ti.f32, shape=(16, 32, 64), offset=(7, 3, -4))   # Works!
+    c = ti.Matrix.field(2, 1, dtype=ti.f32, shape=None, offset=(32,))             # AssertionError
+    d = ti.Matrix.field(3, 2, dtype=ti.f32, shape=(32, 32), offset=(-16, ))       # AssertionError
+    e = ti.field(dt=ti.i32, shape=16, offset=-16)                          # Works!
```
Comment (Rullec) [inline docs/offset.rst:30]:
```suggestion
    e = ti.field(dtype=ti.i32, shape=16, offset=-16)                          # Works!
    f = ti.field(dtype=ti.i32, shape=None, offset=-16)                        # AssertionError
    g = ti.field(dtype=ti.i32, shape=(16, 32), offset=-16)                    # AssertionError
```

File: docs/offset.rst [hunk 8]
```diff
@@ -19,14 +19,14 @@ In this way, the tensor's indices are from ``(-16, 8)`` to ``(16, 72)`` (exclusi
...
+    f = ti.field(dt=ti.i32, shape=None, offset=-16)                        # AssertionError
+    g = ti.field(dt=ti.i32, shape=(16, 32), offset=-16)                    # AssertionError
```
Comment (Rullec) [inline docs/offset.rst:32]:
```suggestion
    e = ti.field(dtype=ti.i32, shape=16, offset=-16)                          # Works!
    f = ti.field(dtype=ti.i32, shape=None, offset=-16)                        # AssertionError
    g = ti.field(dtype=ti.i32, shape=(16, 32), offset=-16)                    # AssertionError
```

File: docs/snode.rst [hunk 9]
```diff
@@ -20,25 +20,25 @@ See :ref:`layout` for more details. ``ti.root`` is the root node of the data str
...
 
-    The following code places two 0-D tensors named ``x`` and ``y``:
+    The following code places two 0-D fields named ``x`` and ``y``:
 
     ::
 
-        x = ti.var(dt=ti.i32)
-        y = ti.var(dt=ti.f32)
+        x = ti.field(dt=ti.i32)
+        y = ti.field(dt=ti.f32)
```
Comment (Rullec) [inline docs/snode.rst:31]:
```suggestion
        x = ti.field(dtype=ti.i32)
        y = ti.field(dtype=ti.f32)
```

File: docs/snode.rst [hunk 10]
```diff
@@ -48,15 +48,15 @@ See :ref:`layout` for more details. ``ti.root`` is the root node of the data str
...
-    :return: (SNode) the structual node where ``tensor`` is placed
+    :parameter a: (ti.field)
+    :return: (SNode) the structual node where ``field`` is placed
 
     ::
 
-        x = ti.var(dt=ti.i32)
-        y = ti.var(dt=ti.f32)
+        x = ti.field(dt=ti.i32)
+        y = ti.field(dt=ti.f32)
```
Comment (Rullec) [inline docs/snode.rst:59]:
```suggestion
        x = ti.field(dtype=ti.i32)
        y = ti.field(dtype=ti.f32)
```

File: docs/snode.rst [hunk 11]
```diff
@@ -106,21 +106,21 @@ Node types
...
+    :parameter shape: (scalar or tuple) shape the field of vectors
     :return: (SNode) the derived child node
 
-    The following code places a 1-D tensor of size ``3``:
+    The following code places a 1-D field of size ``3``:
 
     ::
 
-        x = ti.var(dt=ti.i32)
+        x = ti.field(dt=ti.i32)
```
Comment (Rullec) [inline docs/snode.rst:116]:
```suggestion
        x = ti.field(dtype=ti.i32)
```

File: docs/snode.rst [hunk 12]
```diff
@@ -106,21 +106,21 @@ Node types
...
         ti.root.dense(ti.i, 3).place(x)
 
-    The following code places a 2-D tensor of shape ``(3, 4)``:
+    The following code places a 2-D field of shape ``(3, 4)``:
 
     ::
 
-        x = ti.var(dt=ti.i32)
+        x = ti.field(dt=ti.i32)
```
Comment (Rullec) [inline docs/snode.rst:123]:
```suggestion
        x = ti.field(dtype=ti.i32)
```

**Comment by archibate** [comment]
Wow! That's a lot of change, would you open **another PR** that merge into `rullec-field` to fix this for easy-to-review? @Rullec

**Comment by Rullec** [comment]
Thanks for this reminder, I will do it.

彭于斌 <notifications@github.com> 于 2020年8月2日周日 下午2:00写道：

**Comment by yuanming-hu** [comment]
Thank you both for all the work here. It's extremely well-done!

How about this: let's just apply all the suggestions from @Rullec here in this PR, and @Rullec opens up a new PR with the latest commit. Given most of the PRs in the branch are mainly contributed by @Rullec, the author of this PR should be him, so that the credits are given in a fair manner. (Of course, @archibate should be a Co-authored-by in the new PR.)

If this sounds good to you, I'll take a pass on the new PR instead.

**Comment by Rullec** [comment]
> Thank you both for all the work here. It's extremely well-done!
> 
> How about this: let's just apply all the suggestions from @Rullec here in this PR, and @Rullec opens up a new PR with the latest commit. Given most of the PRs in the branch are mainly contributed by @Rullec, the author of this PR should be him, so that the credits are given in a fair manner. (Of course, @archibate should be a Co-authored-by in the new PR.)
> 
> If this sounds good to you, I'll take a pass on the new PR instead.

Thanks for this constructive idea ;-). If @archibate can be satisfied with this solution, I think it also looks good to me as well.

**Comment by archibate** [comment]
Feel free to close me when you approve and reopen one so that the author will be you :)

**Comment by Rullec** [comment]
> Feel free to close me when you approve and reopen one so that the author will be you :)

Thanks! An equivalant PR1633 has been created.

**Comment by archibate** [comment]
Replaced by #1633.

**Pull request closed without merging**